### PR TITLE
Community, distance, edgescores, generators: cpp coreguideline narrowing conversion

### PR DESCRIPTION
**Pull request by cndolo**
`cppcoreguideline-narrowing-conversion` fixes for the aforementioned modules.

### CODE REVIEW

File: networkit/cpp/edgescores/EdgeScoreLinearizer.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ void EdgeScoreLinearizer::run() {
         for (omp_index pos = 0; pos < static_cast<omp_index>(G->upperEdgeIdBound()); ++pos) {
             edgeid eid = std::get<2>(sorted[pos]);
             if (eid != none) {
-                scoreData[eid] = pos * 1.0 / (G->numberOfEdges()-1); // TODO maybe writing back to sorted and then sorting again by eid could be faster (cache efficiency)?
+                scoreData[eid] = pos * 1.0 / (G->numberOfEdges()-1.0); // TODO maybe writing back to sorted and then sorting again by eid could be faster (cache efficiency)?
```
Comment (fabratu) [inline networkit/cpp/edgescores/EdgeScoreLinearizer.cpp:50]: A static_cast for `pos` and the denominator is needed + maybe a wrap (uint64 -> double).

File: networkit/cpp/community/CoverHubDominance.cpp [hunk 2]
```diff
@@ -56,7 +56,7 @@ void NetworKit::CoverHubDominance::run() {
 
             double dominance = 1;
             if (clusterSizes[i] > 1) {
-                dominance = maxInternalDeg[i] * 1.0 / (clusterSizes[i] - 1);
+                dominance = maxInternalDeg[i] * 1.0 / (clusterSizes[i] - 1.0);
```
Comment (fabratu) [inline networkit/cpp/community/CoverHubDominance.cpp:59]: static_cast around the denominator (+ revert to 1) + static_cast around `maxInternalDeg[i]` (uint64 -> double).

File: networkit/cpp/community/NMIDistance.cpp [hunk 3]
```diff
@@ -42,11 +42,11 @@ double NMIDistance::getDissimilarity(const Graph& G, const Partition& zeta, cons
     std::vector<double> P_eta(eta.upperBound(), 0.0);
 
     for (index C = 0; C < size_zeta.size(); ++C) {
-        P_zeta[C] = size_zeta[C] / (double) n;
+        P_zeta[C] = size_zeta[C] / n;
```
Comment (fabratu) [inline networkit/cpp/community/NMIDistance.cpp:45]: static_cast for `size_zeta` is needed (uint64 -> double)

File: networkit/cpp/community/NMIDistance.cpp [hunk 4]
```diff
@@ -42,11 +42,11 @@ double NMIDistance::getDissimilarity(const Graph& G, const Partition& zeta, cons
...
     }
 
     for (index D = 0; D < size_eta.size(); ++D) {
-        P_eta[D] = size_eta[D] / (double) n;
+        P_eta[D] = size_eta[D] / n;
```
Comment (fabratu) [inline networkit/cpp/community/NMIDistance.cpp:49]: static_cast for `size_zeta` is needed (uint64 -> double)

File: networkit/cpp/community/NMIDistance.cpp [hunk 5]
```diff
@@ -81,11 +81,11 @@ double NMIDistance::getDissimilarity(const Graph& G, const Partition& zeta, cons
             count sizeC = size_zeta[C];
             count sizeD = size_eta[D];
             count sizeO = overlapSizes[O];
-            double factor1 =  sizeO / (double) n;
+            double factor1 =  sizeO / n;
```
Comment (fabratu) [inline networkit/cpp/community/NMIDistance.cpp:84]: static_cast for `size0` is needed (uint64 -> double)

File: networkit/cpp/community/NMIDistance.cpp [hunk 6]
```diff
@@ -81,11 +81,11 @@ double NMIDistance::getDissimilarity(const Graph& G, const Partition& zeta, cons
...
             assert ((sizeC * sizeD) != 0);
             TRACE("overlap of " , C , " and " , D , " has size: " , sizeO);
             TRACE("union of " , C , " and " , D , " has size: " , (sizeD + sizeC - sizeO));
-            double frac2 = (sizeO * n) / (double) (sizeC * sizeD);
+            double frac2 = (sizeO * n) / static_cast<double>(sizeC * sizeD);
```
Comment (fabratu) [inline networkit/cpp/community/NMIDistance.cpp:88]: static_cast for `size0` is needed (uint64 -> double)

File: networkit/cpp/distance/HopPlotApproximation.cpp [hunk 7]
```diff
@@ -126,9 +126,9 @@ void HopPlotApproximation::run() {
             }
         }
         // add nodes that are already connected to all nodes
-        totalConnectedNodes += (G->numberOfNodes() - activeNodes.size()) * G->numberOfNodes();
+        totalConnectedNodes += static_cast<double>((G->numberOfNodes() - activeNodes.size()) * G->numberOfNodes());
```
Comment (fabratu) [inline networkit/cpp/distance/HopPlotApproximation.cpp:129]: `G->numberOfNodes()` is used a lot in the code. Better store the result in a `double`-type variable in the beginning and use this. (All other occurences use the result in expressions with double-variables anyway)

File: networkit/cpp/community/PartitionHubDominance.cpp [hunk 8]
```diff
@@ -49,7 +49,7 @@ void NetworKit::PartitionHubDominance::run() {
 
             double dominance = 1;
             if (clusterSizes[i] > 1) {
-                dominance = maxInternalDeg[i] * 1.0 / (clusterSizes[i] - 1);
+                dominance = maxInternalDeg[i] * 1.0 / (clusterSizes[i] - 1.0);
```
[the remark this answers is not in the document]
Comment (cndolo) [reply networkit/cpp/community/PartitionHubDominance.cpp:52]: Why?

**Comment by cndolo** [comment]
Thanks for reviewing, I have addressed all issues.

**Comment by fabratu** [comment]
LGTM :+1:

**Pull request merged**